src/Constructorio_NET.Tests/client/modules/FacetOptionTest.cs

```diff
@@ -0,0 +1,264 @@
+
+            Assert.AreEqual(facetOption.Value, replacedFacetOption.Value, "Replaced facet option has non-updated Value");
+            Assert.AreEqual(facetOption.DisplayName, replacedFacetOption.DisplayName, "Replaced facet option has non-updated Display Name");
+            Assert.AreEqual(facetOption.Position, replacedFacetOption.Position, "Replaced facet option has non-updated Position");
+            Assert.AreEqual(facetOption.Hidden, replacedFacetOption.Hidden, "Replaced facet option has non-updated Hidden field");
+            Assert.AreEqual(facetOption.Data, replacedFacetOption.Data, "Replaced facet option has non-updated Data field");
+        }
+
+        [Test]
+        public async Task ReplaceFacetOptionDefaults()
+        {
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            FacetOption facetOption = CreateRandomFacetOption();
+            await constructorio.Catalog.CreateFacetOption(facetOption, this.facetGroupName);
+
+            facetOption.DisplayName = null;
+            facetOption.Position = null;
+            facetOption.Hidden = null;
+            facetOption.Data = null;
+
+            FacetOption replacedFacetOption = await constructorio.Catalog.ReplaceFacetOption(facetOption, this.facetGroupName);
+
+            Assert.AreEqual(facetOption.Value, replacedFacetOption.Value, "Replaced facet option has incorrect Value");
+            Assert.AreEqual(null, replacedFacetOption.DisplayName, "Replaced facet option has non-default Display Name");
+            Assert.AreEqual(null, replacedFacetOption.Position, "Replaced facet option has non-default Position");
+            Assert.AreEqual(false, replacedFacetOption.Hidden, "Replaced facet option has non-default Hidden field");
+            Assert.AreEqual(null, replacedFacetOption.Data, "Replaced facet option has non-default Data field");
+        }
+
+        [Test]
+        public async Task PartiallyUpdateFacetOption()
+        {
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            FacetOption facetOption = CreateRandomFacetOption();
+            System.Collections.Hashtable data = new System.Collections.Hashtable();
+            data.Add("hello", 3);
+            facetOption.Data = data;
+            facetOption.Position = 2;
+            facetOption.Hidden = false;
+
+            await constructorio.Catalog.CreateFacetOption(facetOption, this.facetGroupName);
+
+            facetOption.DisplayName = "updated";
+            facetOption.Position = null;
+            facetOption.Hidden = null;
+            facetOption.Data = null;
+
+            FacetOption updatedFacetOption = await constructorio.Catalog.PartiallyUpdateFacetOption(facetOption, this.facetGroupName);
+
+            Assert.AreEqual(facetOption.DisplayName, updatedFacetOption.DisplayName, "Non-updated Display Name");
```

Comment:
Should the message here be `"Display name should have been updated."` instead?

Reply:
Good eye, that's a clearer phrasing. Thanks for the catch, I'll update similar messages to match.